support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -224,16 +224,37 @@ function getAmountsTestParticipations(
 		return urlTest;
 	}
 
-	const { test } = settings.amounts?.[countryGroupId] ?? {};
+	if (!settings.amounts) {
+		return null;
+	}
+
+	const AMOUNTS = settings.amounts;
+
+	// Two-tier amounts - check for live country test, else get a region test
+	let targetTest = AMOUNTS.find((t) => t.target === country && t.isLive);
+	if (!targetTest) {
+		targetTest = AMOUNTS.find((t) => t.target === countryGroupId);
+	}
+
+	if (!targetTest) {
+		return null;
+	}
+
+	const { testName, liveTestName, isLive, seed, variants } = targetTest;
+
+	if (!variants.length) {
+		return null;
+	}
 
-	if (!test || !test.isLive) {
+	if (!isLive || variants.length === 1) {
```

Comment:
The array of amounts tests generated in RRCP, stored in S3 and retrieved by SDC/SF will always include one-and-only-one test for each region (countryGroup). For these region tests there will always be a `control` variant, alongside 0-or-more AB test variants. Region tests are "always on"; the `isLive` boolean in these tests tells us whether an AB test is running (true) in which case there will be a choice to be made about which variant the user falls into, or whether there's no AB test running in which case the user should see the control variant. Every region test will have an obligatory "control" variant which is always the first variant in the test variants array.

This differs for country-specific tests. The `isLive` boolean in these tests tells us whether to use the country-specific data (true) in an AB test, or to fall back to that country's region test (false).

The purpose of the code here is to determine whether there's an active AB amounts test running for that user, given their geographical location (country or region). If there is an active AB test then the test name/variant gets coded as a key:value pair into an object which then amalgamates into a `participations` object which includes details of all the current (often hardcoded) tests being run for that user across the entire support site.

Note that this code is very similar to the code in the sibling `helpers.ts` file, which is where the data actually gets returned to the SF `priceCards` component. I have no idea why our predecessors chose to do things this way, but I am not going to argue with their wisdom!

Reply:
I would argue strongly against that - it is what we had before and what this series of PRs is moving away from. Also, there is no guarantee that regions will remain static going forward and it makes sense to me to keep hardcoded regions as far away as possible from the data model. Does that make sense?

Reply:
> I would argue strongly against that - it is what we had before and what this series of PRs is moving away from. Also, there is no guarantee that regions will remain static going forward and it makes sense to me to keep hardcoded regions as far away as possible from the data model. Does that make sense?

I don't follow I'm afraid! I just meant that a JS object feels like a more appropriate data structure than an array if you want to enforce some kind of uniqueness, and if you're often looking up that item in the data structure based on a given property. But happy to continue to conversation offline as it doesn't sound like this is something that'll change in this PR.

Reply:
I think an array is the correct choice, despite the additional complexity.
As Rik says, we're moving away from a simple one-test-per-region lookup, to something with more sophisticated targeting.
Though I wonder if the ordering of the array of tests should be used, instead of scanning the array twice. And the RRCP puts regional tests last in the priority order. In theory other types of targeting would then be possible in future.